### DISCUSSION TITLE
Compile fails after upgrading to v4.0(undefined reference to `_Z9setupMenuv')

### DISCUSSION THREAD

**Discussion by prowong** [davetcc/tcMenu#376] [Q&A] [answered]
All things work fine in v3.1. But after upgrading to v4.0, I just created a very simple menu with only two items, and the compilation error message is as follows. I've tried a full clean install with the same result. On the other hand, when opening the old version of the project in v4.0 and generating code, the compilation is successful. Any suggestions?

`d:/program files/arduinodata/packages/esp8266/tools/xtensa-lx106-elf-gcc/3.1.0-gcc10.3-e5f9fec/bin/../lib/gcc/xtensa-lx106-elf/10.3.0/../../../../xtensa-lx106-elf/bin/ld.exe: C:\Users\Patrick\AppData\Local\Temp\arduino\sketches\3797C9C34B5C4AC64A24480549BFE714\sketch\VNC008.ino.cpp.o:(.text.setup+0x0): undefined reference to `_Z9setupMenuv'
collect2.exe: error: ld returned 1 exit status`

**Comment by prowong** [comment, marked as answer]
I found a solution. As shown in the figure below, change this setting to be the same as v3.1. Is this a compiler compatibility issue?
![20230710150653](https://github.com/davetcc/tcMenu/assets/8499801/82fd539f-1d63-4aca-8b35-550e74fe9eb8)

**Reply by davetcc** [reply]
Thanks for determining that. I’ve not had chance to look at tcMenu for the past few weeks. But I suspect that the generated folder option is only compatible with platformio, mbed and cmake. I’ll switch the default back to all in one folder in the next release.

**Reply by davetcc** [reply]
https://github.com/davetcc/tcMenu/issues/377 to set the default back to all in one dir.

**Reply by davetcc** [reply]
BTW this was done in 4.1, the default is now back to Arduino-supported options.
